diff --git a/src/stdgpu/cuda/impl/device.cpp b/src/stdgpu/cuda/impl/device.cpp
--- a/src/stdgpu/cuda/impl/device.cpp
+++ b/src/stdgpu/cuda/impl/device.cpp
@@ -57,6 +57,9 @@ print_device_information()
         return;
     }
 
+    int clock_rate_khz = 0;
+    STDGPU_CUDA_SAFE_CALL(cudaDeviceGetAttribute(&clock_rate_khz, cudaDevAttrClockRate, 0));
+
     std::size_t free_memory = 0;
     std::size_t total_memory = 0;
     STDGPU_CUDA_SAFE_CALL(cudaMemGetInfo(&free_memory, &total_memory));
@@ -77,7 +80,7 @@ print_device_information()
     printf("+---------------------------------------------------------+\n");
     printf("| Compute Capability        :   %1d.%1d                       |\n", properties.major, properties.minor);
     printf("| Clock rate                :   %-6.0f MHz                |\n",
-           static_cast<double>(detail::kilo_to_mega_hertz(static_cast<float>(properties.clockRate))));
+           static_cast<double>(detail::kilo_to_mega_hertz(static_cast<float>(clock_rate_khz))));
     printf("| Global Memory             :   %-6.3f GiB / %-6.3f GiB   |\n",
            static_cast<double>(detail::byte_to_gibi_byte(static_cast<float>(free_memory))),
            static_cast<double>(detail::byte_to_gibi_byte(static_cast<float>(total_memory))));
